Samples/iOS-Swift/iOS-Swift/ViewController.swift

```diff
@@ -126,19 +131,53 @@ class ViewController: UIViewController {
         }
     }
     
-    @IBAction func captureNSException(_ sender: Any) {
+    @IBAction func captureNSException(_ sender: UIButton) {
+        highlightButton(sender)
         let exception = NSException(name: NSExceptionName("My Custom exeption"), reason: "User clicked the button", userInfo: nil)
         let scope = Scope()
         scope.setLevel(.fatal)
         // !!!: By explicity just passing the scope, only the data in this scope object will be added to the event; the global scope (calls to configureScope) will be ignored. If you do that, be careful–a lot of useful info is lost. If you just want to mutate what's in the scope use the callback, see: captureError.
         SentrySDK.capture(exception: exception, scope: scope)
     }
     
-    @IBAction func captureFatalError(_ sender: Any) {
+    @IBAction func captureFatalError(_ sender: UIButton) {
+        highlightButton(sender)
         fatalError("This is a fatal error. Oh no 😬.")
     }
+
+    var span: Span?
+    let profilerNotification = NSNotification.Name("SentryProfileCompleteNotification")
     
-    @IBAction func captureTransaction(_ sender: Any) {
+    @IBAction func startTransaction(_ sender: UIButton) {
+        highlightButton(sender)
+        guard span == nil else { return }
+        span = SentrySDK.startTransaction(name: "Manual Transaction", operation: "Manual Operation")
+
+        NotificationCenter.default.addObserver(forName: profilerNotification, object: nil, queue: nil) { note in
```

Comment:
`m`: I think it would be better to subscribe to this in `viewWillAppear` and unsubscribe in `viewWillDisappear`. Otherwise, we add an observer every time you click the button.

Reply:
Yeah I think this would be fine.

Reply:
Actually I did this so that we don't get an arbitrary amount of unexpected alerts popping up for e.g. a profiled launch span, so i think we should leave as-is. I have a guard on 153 that prevents multiple subscriptions.